app/connectors/SoftDrinksIndustryLevyConnector.scala

```diff
@@ -32,6 +32,23 @@ class SoftDrinksIndustryLevyConnector @Inject()(
 
   lazy val sdilUrl: String = frontendAppConfig.sdilBaseUrl
 
+  private def getRosmRegistration(utr: String):String = s"$sdilUrl/rosm-registration/lookup/$utr"
+
+
+  def retreiveRosmSubscription(utr: String, internalId: String)
+                              (implicit hc: HeaderCarrier): Future[Option[RosmRegistration]] = {
+    sdilSessionCache.fetchEntry[OptRosmRegistration](internalId, SDILSessionKeys.ROSM_REGISTRATION).flatMap{
+      case Some(optRosmRegistration) => Future.successful(optRosmRegistration.optRosmRegistration)
+      case None =>
+        http.GET[Option[RosmRegistration]](getRosmRegistration(utr)).flatMap {
+          optRosmRegistration =>
+            sdilSessionCache.save(internalId, SDILSessionKeys.ROSM_REGISTRATION, OptRosmRegistration(optRosmRegistration))
+              .map{_ =>
+                optRosmRegistration}
+        }
+    }
+  }
```

Comment:
As the user fails if they don't have a rosmRegistration we should just save to cache if it is present. Especially given soon we will be adding logic which allows the user to choose a different utr if they don't have rosm for the utr in the session.